app/download/views.py

```diff
@@ -93,3 +98,40 @@ def download_document_b64(service_id, document_id):
     response.headers["X-Robots-Tag"] = "noindex, nofollow"
 
     return response
+
+
+@download_blueprint.route(
+    "/services/<uuid:service_id>/documents/<uuid:document_id>/scan-verdict", methods=["POST"]
+)
+def check_scan_verdict(service_id, document_id):
```

Comment:
Definitely a fair point - I have it set up so that notification-api will make a request to this endpoint before downloading the file. But it is probably more foolproof to do it the way you are suggesting. I will give that a try.

Reply:
Actually on second thought - I did have it set up like you are suggesting before, and the reason I moved to a single endpoint was because we have 2 endpoints for downloading files - one for "attach" mode and one for "link" mode. For link mode, the download endpoint is not actually called by notification-api, only by the user who downloads the file. So we need some kind of dedicated "check scan status" endpoint to prevent link mode from sending viruses. 

That being said, I could keep this new endpoint but also call `check_scan_verdict` from the other 2 "download document" endpoints as another layer of safety.

Reply:
Makes sense to me!

Reply:
Okay as @andrewleith and I discussed offline - I will make a follow up PR to add the check to the 2 doc download endpoints, and that can be merged after changes to the notification-api are merged. That way we can merge this PR and still not cause compatibility problems with the current notification-api.